pkg/distributor/distributor.go

```diff
@@ -691,6 +695,23 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 	return &cortexpb.WriteResponse{}, firstPartialErr
 }
 
+func (d *Distributor) GetShuffleShardingRing(ctx context.Context) (ring.ReadRing, error) {
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	//Get any overrides for the current tenant
+	limits := d.limits.GetOverridesForUser(userID)
+	//Retrieves all the current ingesters rings.
+	subRing := d.ingestersRing
+	//Determines if Shuffle Sharding is enabled and or the overrides has a Sharding Strategy of greater than 0.
+	if d.cfg.ShardingStrategy == util.ShardingStrategyShuffle && limits.IngestionTenantShardSize > 0 {
```

Comment:
This seems unnecessary. Even if you pass `limits.IngestionTenantShardSize <= 0`, `ShuffleShard` still returns the whole ring, which is the same as this behavior.

Reply:
I think we can still keep the code at L680.

Reply:
Ahh yes, I should have went further down the call stack! Sorry about that:

Future readers - what I believe @yeya24  is referring to the below block which will just return the entire ring if the size is 0. 
```go
func (r *Ring) shuffleShardWithCache(identifier string, size int, zoneStableSharding bool) ReadRing {
	// Nothing to do if the shard size is not smaller than the actual ring.
	if size <= 0 || r.InstancesCount() <= size {
		return r
}
```